src/lib.rs

```diff
@@ -100,21 +100,30 @@ pub fn run_main(
 #[cfg(test)]
 mod run_main {
     use super::*;
+    use executable_mock::create_mock_executable;
     use std::io::Cursor;
     use test_utils::TempFile;
 
     #[test]
     fn when_passed_executable_mock_flag_behaves_like_executable_mock() -> R<()> {
-        let file = TempFile::write_temp_script("first line\nsecond line\n")?;
+        let context = Context::new_test_context();
+        let executable_contents = create_mock_executable(
+            &context,
+            executable_mock::Config {
+                stdout: b"foo".to_vec(),
+                exitcode: 0,
+            },
+        )?;
+        let file = TempFile::write_temp_script(&executable_contents)?;
         let args = vec![
             "executable-mock".to_string(),
             "--executable-mock".to_string(),
```

Comment:
Yes, it is. However users are not ever supposed to pass this in.

This is a bit wonky. `check-protocols` needs to be able to mock calls to `execv`. It accomplishes that by creating a temporary file, that is an executable and switching out the path of the executed subprocess before entering the `execve` syscall, setting it to that temporary file. The temporary file looks something like this:

```
#!<path to the currently running check-protocols executable>/check-protocols --executable-mock
<binary serialization of a executable_mock::Config>
```

And then when that flag is passed into `check-protocols` -- instead of running normally -- it reads the `executable_mock::Config` from the given file and acts as the executable mock.

As I said this is a bit wonky, but it has two nice properties:

- `check-protocols` can be installed as one single, self-contained executable. No need for another executable mocking helper tool.
- The mock executables and the calling `check-protocols` are guaranteed to always be in sync. Meaning they're always compiled from the exact same source code version, since they are the same executable.

Reply:
the explanation makes sense to me. i don't think it would hurt to mention this in the readme, or in a comment either here or in the `executable_mock` file.